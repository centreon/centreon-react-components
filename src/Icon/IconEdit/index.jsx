import React from 'react';

import { styled } from '@material-ui/core/styles';
import Edit from '@material-ui/icons/Edit';

import MaterialIcon from '../MaterialIcon';

const FloatingEdit = styled(Edit)(() => ({
  color: '#0072CE',
  fontSize: 21,
  position: 'absolute',
  right: 3,
}));

function IconEdit(props) {
  return (
    <MaterialIcon {...props}>
      <FloatingEdit />
    </MaterialIcon>
  );
}

export default IconEdit;
