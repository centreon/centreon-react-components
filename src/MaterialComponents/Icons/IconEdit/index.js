import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import Edit from '@material-ui/icons/Edit';

const useStyles = makeStyles(theme => ({
  root: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  icon: {
    color: '#0072CE',
    cursor: 'pointer',
    fontSize: 21,
    position: 'absolute',
    right: 3,
  },
}));

function IconEdit({customStyle, onClick}) {
  const classes = useStyles();

  return (
    <Edit onClick={onClick} style={customStyle} className={classes.icon} />
  );
}

export default IconEdit;
