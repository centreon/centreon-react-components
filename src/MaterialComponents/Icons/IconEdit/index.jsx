import React from 'react';
import Edit from '@material-ui/icons/Edit';
import styled from '@emotion/styled';
import MaterialIcon from '../MaterialIcon';

const FloatingEdit = styled(Edit)(() => ({
  color: '#0072CE',
  fontSize: 21,
  position: 'absolute',
  right: 3,
}));

function IconEdit(props) {
  return (
    <MaterialIcon {...props}>
      <FloatingEdit />
    </MaterialIcon>
  );
}

export default IconEdit;
