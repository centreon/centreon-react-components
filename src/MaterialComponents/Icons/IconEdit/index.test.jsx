/* eslint-disable no-undef */

import React from 'react';
import { create } from 'react-test-renderer';
import IconEdit from '.';

describe('IconEdit', () => {
  it('renders', () => {
    const wrapper = create(<IconEdit />);

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
