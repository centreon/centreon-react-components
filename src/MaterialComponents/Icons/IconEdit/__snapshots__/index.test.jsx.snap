// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconEdit renders 1`] = `
.emotion-1 {
  display: inline-block;
  vertical-align: middle;
  height: 24px;
  cursor: pointer;
  color: #707070;
}

.emotion-0 {
  color: #0072CE;
  font-size: 21px;
  position: absolute;
  right: 3px;
}

<span
  className="emotion-1"
>
  <svg
    aria-hidden="true"
    className="MuiSvgIcon-root emotion-0"
    focusable="false"
    role="presentation"
    viewBox="0 0 24 24"
  >
    <path
      d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
    />
    <path
      d="M0 0h24v24H0z"
      fill="none"
    />
  </svg>
</span>
`;
